### ISSUE TITLE
too many open files

### ISSUE THREAD

**Issue by jimpick** [web3-storage/web3.storage#341]
Using the put-files.js script from the tutorial:

```
$ ./archive.sh 
node put-files.js --token=$WEB3_STORAGE_API_TOKEN ../estuary-archive
Uploading 8992 files
node:events:371
      throw er; // Unhandled 'error' event
      ^

Error: EMFILE: too many open files, open '/Users/jim/projects-jpimac/spark/miner-power/estuary-archive/miner-power/power-994867.json'
Emitted 'error' event on ReadStream instance at:
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -24,
  code: 'EMFILE',
  syscall: 'open',
  path: '/Users/jim/projects-jpimac/spark/miner-power/estuary-archive/miner-power/power-994867.json'
}
```

**Comment by alanshaw** [comment]
@vasco-santos would you mind also looking into this? 🙏

**Comment by vasco-santos** [comment]
wow, this is strange. Sure, I will add it to my list

**Comment by atopal** [comment]
probably related: https://github.com/web3-storage/web3.storage/issues/318

**Comment by vasco-santos** [comment]
An update on https://github.com/web3-storage/ipfs-car/issues/69#issuecomment-908514767 as both seem related.

**Comment by atopal** [comment]
related: https://github.com/web3-storage/ipfs-car/issues/69

**Comment by vasco-santos** [comment]
@jimpick are you able to re-test this with the newest versions? 

This https://github.com/web3-storage/ipfs-car/issues/69#issuecomment-986848832 should have helped, but I also need to do more tests around other improvements

**Comment by olizilla** [comment]
Thanks for reporting Jim! There is a [fix](https://github.com/web3-storage/files-from-path/pull/18) for the issue that landed in `files-for-path@v0.2.5` that is re-exported by the `web3.storage`. 

Running `npm install web3.storage@latest` in same folder that your package.json file is for your copy of `put-files.js` should fix the issue.
